app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -688,4 +705,27 @@ open class MainPresenterImpl(val mapzenLocation: MapzenLocation, val bus: Bus,
     mainViewController?.drawRoute(route)
     mainViewController?.hideProgress()
   }
+
+  private fun featuresExist(features: List<Feature>?): Boolean {
+    return features != null || features!!.size > 0
```

Comment:
Should this be `&&` operator?

Currently the second half of conditional will never evaluate unless features is null in which case the `!!` operator will throw a NPE.